blackjax/mcmc/ghmc.py

```diff
@@ -0,0 +1,254 @@
+
+    Returns
+    -------
+    A kernel that takes a rng_key, a Pytree that contains the current state
+    of the chain, and free parameters of the sampling mechanism; and that
+    returns a new state of the chain along with information about the transition.
+    """
+
+    _, kinetic_energy_fn, _ = metrics.gaussian_euclidean(jnp.ones(1))
+    sample_proposal = proposal.nonreversible_slice_sampling
+
+    def one_step(
+        rng_key: PRNGKey,
+        state: GHMCState,
+        logprob_fn: Callable,
+        step_size: PyTree,  # float,
+        alpha: float,
+        delta: float,
+        logprob_grad_fn: Optional[Callable] = None,
+    ) -> Tuple[GHMCState, hmc.HMCInfo]:
+        """Generate new sample with the Generalized HMC kernel.
+
+        Parameters
+        ----------
+        rng_key
+            JAX's pseudo random number generating key.
+        state
+            Current state of the chain.
+        logprob_fn
+            (Unnormalized) Log density function being targeted.
+        step_size
+            Pytree with the same structure as the targeted position variable
+            specifying the step size used for each dimension of the target.
+        alpha
+            Variable specifying the degree of persistent momentum, complementary
+            to independent new momentum.
+        delta
+            Fixed (non-random) amount of translation added at each new iteration
+            to the slice variable for non-reversible slice sampling.
+        logprob_grad_fn
+            Optional function customizing the gradients of the target log density.
+        """
+
+        def potential_fn(x):
+            return -logprob_fn(x)
+
+        symplectic_integrator = velocity_verlet(
+            potential_fn, kinetic_energy_fn, logprob_grad_fn
+        )
+        proposal_generator = hmc.hmc_proposal(
+            symplectic_integrator,
+            kinetic_energy_fn,
+            step_size,
+            divergence_threshold=divergence_threshold,
+            sample_proposal=sample_proposal,
+        )
+
+        key_momentum, key_noise = jax.random.split(rng_key)
+        position, momentum, potential_energy, potential_energy_grad, slice = state
+        # New momentum is persistent
+        momentum = update_momentum(key_momentum, position, momentum, alpha)
+        # Slice is non-reversible
+        slice = ((slice + 1.0 + delta + noise_gn(key_noise)) % 2) - 1.0
+
+        integrator_state = integrators.IntegratorState(
+            position, momentum, potential_energy, potential_energy_grad
+        )
+        proposal, info = proposal_generator(slice, integrator_state)
+        proposal = hmc.flip_momentum(proposal)
+        state = GHMCState(
+            proposal.position,
+            proposal.momentum,
+            proposal.potential_energy,
+            proposal.potential_energy_grad,
+            info.acceptance_probability,
+        )
+
+        return state, info
+
+    return one_step
+
+
+def velocity_verlet(
```

Comment:
That you specify a step size per component instead of a mass matrix is unfortunate, but this will do for now.